Compass/CompassViewController.swift

```diff
@@ -123,20 +157,21 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
                 //now it should point towards the target location in case the device points towards north -> test this
                 self.ArrowImage.transform = CGAffineTransform(rotationAngle: CGFloat(dirRadiant - headingR))
             }
-            degreeLabel.text = String(dirRadiant)
         }
     }
     func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
         
         currentLocation = locations[0]
         myUserData.location = locations[0]
-            if par.updateUserLocation(classN: "Users", uData: myUserData) {
-                print("Current object id = \(myUserData.objectID)")
-            } else {
-                if let tempID = UserDefaults.standard.string(forKey: "parseObjectID") {
-                myUserData.objectID = tempID
-                    }
-            }
+        par.updateUserLocation(classN: "Locations", uData: myUserData)
+        
+        guard let targetLocation = targetLocation else {
+            return
+        }
+        guard let distance = currentLocation?.distance(from: targetLocation) else {
+            return
+        }
+        distanceLabel.text = "\(Int(distance))m"
```

Comment:
This is great! 
Definitely a matter of opinion, but I just want to let you know you can chain guard let statements, even though the second one needs `targetLocation`, you can combine them both into one guard, like this:
``` 
guard let targetLocation = targetLocation, let distance = currentLocation?.distance(from: targetLocation) else {
    return
}
```
(and for readability you can have the second one on a different line:
```
guard let targetLocation = targetLocation, 
      let distance = currentLocation?.distance(from: targetLocation) else {
    return
}
```
But, like I said, some people would prefer to keep it seperate like you have anyway.